### ISSUE TITLE
Broken proxy for .local domains

### ISSUE THREAD

**Issue by pmishev** [symfony-cli/symfony-cli#288]
I'm opening a separate issue for my question I asked in https://github.com/symfony-cli/symfony-cli/issues/284#issuecomment-1455976330, as the OP closed that one.

To sum it up, after https://github.com/symfony-cli/symfony-cli/commit/7c2f9a7c00cc716f0c73d18fc24fb567a6fd2458 by @tucksaun , `.local` domains now do not work with the local proxy, at least in Chrome.

The change introduced now first checks if the domain is resolvable, and for a reason unknown to me, Chrome takes ~5s to decide that it is not. You can try at `chrome://net-internals/?#dns`

If you try to resolve `blah.wip`, it decides it's not resolvable instantly, so all is fine.
But if you try to resolve `blah.local`, it takes 5s until it figures out it's not resolvable.

This cannot even be worked around by adding in `/etc/hosts`:
```
127.0.0.1 blah.local
```
as the request will not use the proxy at all.

**Comment by tucksaun** [comment]
> for a reason unknown to me, Chrome takes ~5s to decide that it is not

This is because `.local` is a special case - but still valid - tld from the official list and is used for mDNS resolution.

Alternatively to what I said in https://github.com/symfony-cli/symfony-cli/issues/284#issuecomment-1458883863 we could also check if the tld is within IANA nor https://en.wikipedia.org/wiki/List_of_Internet_top-level_domains#Special-Use_Domains list.
This check could be done during proxy start to be done only once.
Maybe only the special use list needs to be actually checked because otherwise resolutions should be fast enough?

Do you think that could work for you?

**Comment by pmishev** [comment]
Perhaps that would work. However, I have to say that having realized the problem, we decided to just switch to using .wip instead of .local.
It may be beneficial though to at least add a line in the docs warning against using .local, if nothing else.
